render/id.go

```diff
@@ -62,15 +59,24 @@ func NewDerivedExternalNode(n report.Node, addr string, local report.Networks) (
 	return report.Node{}, false
 }
 
-// DNSNames returns a prioritized list of snooped and reverse-resolved
-// DNS names associated with node n.
-func DNSNames(n report.Node) []string {
-	snoopedNames, _ := n.Sets.Lookup(endpoint.SnoopedDNSNames)
-	reverseNames, _ := n.Sets.Lookup(endpoint.ReverseDNSNames)
-	// sort the names, to make selection for display more
+// DNSFirstMatch returns the first DNS name where match() returns
+// true, from a prioritized list of snooped and reverse-resolved DNS
+// names associated with node n.
+func DNSFirstMatch(n report.Node, match func(name string) bool) (string, bool) {
```

Comment:
D'oh.

Reply:
Making this higher-order seems overkill. We could call the function `FirstKnownService`.

Reply:
You see it's called differently in two places?